lib/code_ownership/private/glob_cache.rb

```diff
@@ -32,6 +32,40 @@ def raw_cache_contents
         @raw_cache_contents
       end
 
+      sig { params(files: T::Array[String]).returns(FilesByMapper) }
+      def mapper_descriptions_that_map_files(files)
+        files_by_mappers = T.let({}, FilesByMapper)
+
+        # When looking at many files, expanding the cache out using Dir.glob and checking for intersections is faster
+        # TODO: optimize this number
+        if files.count > 100
```

Comment:
Good question about how to optimize this 🤔 

It should be possible to write up a script to test different numbers of them, and let them run a overnight 😀  

Practically speaking, we care about when the pre-commit is called, which calls this with a list of changed files. If there's a ton of files, I don't think this will necessarily be the slow thing.